### PR TITLE
Update Amazon and AWS

### PR DESCRIPTION
**Pull request by fainpablo**
Added 'phone' to Amazon.

amazon-2fa-options

### CODE REVIEW

**Review by Carlgo11 — APPROVED** [review]

**Comment by phallobst** [comment]
Does this apply only to the "Retail" entry of Amazon?

**Comment by Carlgo11** [comment]
@phallobst no it applies to any oauth login for Amazon (including AWS)

**Comment by fainpablo** [comment]
I'm updating Amazon Pay now. Phone does not apply to AWS nor AWS Workspaces.

I've double checked on my AWS account and confirmed they accept both U2F and other hardware tokens (one at a time), however they're dropping SMS support (https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_mfa_enable_sms.html). 

Should we remove SMS from AWS?

**Comment by phallobst** [comment]
> AWS will soon end support for SMS

_Valve "soon" or normal "soon"?_ (:copyright: @Carlgo11)

I'd say remove the "sms" method from the AWS entry.

**Comment by Carlgo11** [comment]
AWS actually uses two different 2FA steps. The first one is for authenticating the Amazon account and the second is any AWS specific 2FA.

So if you setup 2FA for your amazon account and then at the account settings in AWS you’ll have to authenticate with two TOTP codes.

Thereby all Amazon 2FA options apply to AWS but not all AWS options apply to your Amazon account.

**Comment by phallobst** [comment]
So they discontinue SMS as a 2FA method for AWS, probably because of the security implications, but are fine with using it for the retail market? What a considerate company!

**Comment by fainpablo** [comment]
LOL Yes. Not to mention you can only setup one 2FA method at a time in AWS. If you add a security key and lose it, you'll in trouble.

So I've added "phone" and maintained "sms" from AWS, considering users need to go through Amazon 2FA challenge before getting into AWS.

File: _data/cloud.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ websites:
     img: aws.png
     tfa:
       - sms
+      - phone
       - totp
```
Comment (jamcat22) [inline _data/cloud.yml]:
```suggestion
      - totp
```

**Review by jamcat22 — CHANGES_REQUESTED** [review]
I'd suggest this be handled similar to Atlassian Cloud and Bitbucket, where Atlassian lists the methods for Atlassian's login and Bitbucket lists the methods for Bitbucket's login, even though Bitbucket [uses Atlassian Accounts for login](https://confluence.atlassian.com/bitbucket/upgrade-to-atlassian-account-829056056.html).

Bitbucket works similar to AWS, where you first log in to your Atlassian Account with 2FA if enabled, then provide 2FA for Bitbucket if enabled. While Atlassian supports SMS and TOTP, Bitbucket instead supports U2F and TOTP. When enabling Atlassian 2FA, you inherently protect the main access method to your Bitbucket account through your Atlassian web login. However, unless 2FA is enabled on the Bitbucket account itself, Atlassian 2FA doesn't protect basic authentication when using the API. Bitbucket 2FA must be enabled separately to force the use of [application-specific passwords](https://confluence.atlassian.com/bitbucket/app-passwords-828781300.html) to protect basic authentication logins on the API. It's also possible to enable Bitbucket 2FA without enabling Atlassian 2FA (for example, if you're in a Bitbucket team that [enforces 2FA](https://confluence.atlassian.com/bitbucket/control-access-to-your-team-s-private-content-915147488.html#Controlaccesstoyourteam'sprivatecontent-Requiringtwo-stepverification)).

Amazon and AWS security both work similarly. When enabling Amazon 2FA, you inherently protect the main access method to your AWS account, though your Amazon login. However, in the same vein as Bitbucket, you can leave basic authentication methods unprotected by Amazon 2FA. Also, AWS 2FA becomes much more important when using [Cross-Account Access](https://aws.amazon.com/blogs/security/how-do-i-protect-cross-account-access-using-mfa-2/) and [IAM Users](https://docs.aws.amazon.com/IAM/latest/UserGuide/id.html#id_iam-users), users that login to AWS directly, **without** going through Amazon's login system. For IAM Users, because authentication does not go through Amazon's system first, they are not able to use the methods available to main Amazon account logins and must use TOTP, hardware, or U2F with their AWS specific 2FA login.

**Comment by fainpablo** [comment]
PR replaced by #4736, #4737, and #4738.

**Pull request closed without merging**